Source/Evolve/WarpXEvolve.cpp

```diff
@@ -929,3 +938,46 @@ WarpX::applyMirrors(Real time){
         }
     }
 }
+
+void
+WarpX::CheckSignals()
+{
+    // We assume that signals will definitely be delivered to rank 0,
+    // and may be delivered to other ranks as well. For coordination,
+    // we process them according to when they're received by rank 0.
+    if (amrex::ParallelDescriptor::MyProc() == 0) {
+        for (int i = 0; i < NUM_SIGNALS; ++i) {
+            // Read into a local temporary to ensure the same value is
+            // used throughout. Atomically exchange it with false to
+            // unset the flag without risking loss of a signal - if a
+            // signal arrives after this, it will be handled the next
+            // time this function is called.
+            bool signal_i_received = signal_received_flags[i].exchange(false);
+
+            if (signal_i_received) {
+                signal_actions_requested[SIGNAL_REQUESTS_BREAK] |= signal_conf_requests_break[i];
+                signal_actions_requested[SIGNAL_REQUESTS_CHECKPOINT] |= signal_conf_requests_checkpoint[i];
+            }
+        }
+    }
+
+#if defined(AMREX_USE_MPI)
+    auto comm = amrex::ParallelDescriptor::Communicator();
+    MPI_Ibcast(signal_actions_requested, SIGNAL_REQUESTS_MAX+1, MPI_CXX_BOOL, 0, comm, &signal_mpi_ibcast_request);
```

Comment:
Is there some common MPI error checking wrapper I should use for these one-off calls?